public/app/plugins/datasource/prometheus/components/PromLink.tsx

```diff
@@ -18,6 +18,14 @@ interface State {
 export default class PromLink extends Component<Props, State> {
```

Comment:
Would it be too much to ask to refactor this component to a function component? 😄 

2 things I noticed
- the same logic is used in componentDidMount and componentDidUpdate
- getExternalLink is async because?

Reply:
No, not much work at all. And you are totally right ⭐️ - leave the code in better shape than.... So I refactored it to hooks + as I was adding tests, I switched to react-testing-library. 🙂

Reply:
Also async was not necessary.